### PR TITLE
[13.0][FIX] dms: Set res_mimetype correctly in files

### PR DESCRIPTION
**Pull request by victoralmau**
Set `res_mimetype` correctly in files.

Please @Yajo and @pedrobaeza can you review it?

@Tecnativa TT31461

### CODE REVIEW

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
I'm not sure we really need this, as this is very partial, but deploy properly a library that detects all the possible mimetypes instead of the limited implementation inside Odoo. Please check this code:

https://github.com/odoo/odoo/blob/c5a22a122eccd18247280a941efd2c96935c43d6/odoo/tools/mimetypes.py#L165-L179

Try in your local to put such library to see if it's detected properly, and if so we should:

- Add in base Doodba such library
- Put in the README in the installation section the recommendation to install the library.

File: dms/models/dms_file.py [hunk 1]
```diff
@@ -377,10 +378,15 @@ def _compute_extension(self):
         for record in self:
             record.extension = file.guess_extension(record.name)
 
-    @api.depends("content")
+    @api.depends("content", "name")
     def _compute_mimetype(self):
+        """ Similar to addons/base/models/ir_attachment.py#L232 """
```
Comment (yajo) [inline dms/models/dms_file.py]: This comment says nothing. And the line number can change. Please add a proper docstring and, if you need to refer to another resource, add a permalink.

File: dms/models/dms_file.py [hunk 2]
```diff
@@ -377,10 +378,15 @@ def _compute_extension(self):
...
         for record in self:
-            record.res_mimetype = guess_mimetype(base64.b64decode(record.content or ""))
+            record.res_mimetype = mimetypes.guess_type(record.name or "")[0]
+            if record.res_mimetype == "application/octet-stream":
+                record.res_mimetype = guess_mimetype(
+                    base64.b64decode(record.content or "")
+                )
```
Comment (yajo) [inline dms/models/dms_file.py]: IIUC, you're trying to get the mimetype from the file name and, later, from the file contents.

IMHO it should be the opposite. The contents are usually more reliable than the file extension to know the kind of file you have.

**Review by yajo — CHANGES_REQUESTED** [review]
As @pedrobaeza said, the first thing to try is to install those libraries and see if the problem goes away automatically. If that happens, close this.

Otherwise, a couple of code comments below.

**Comment by victoralmau** [comment]
I tested locally to add the `python-magic` library and this changes will not be necessary because the `res_mimetype` field will be defined correctly.

Now the question is, do we add it as a dependency in the manifest or do we suggest its installation in the readme file and skip the parts of the test that are necessary if the library is not added?

Let's not forget that this happened because the `res_mimetype` field for `.mp4` files was not defined correctly.

**Comment by pedrobaeza** [comment]
See my previous comment in https://github.com/OCA/dms/pull/131#pullrequestreview-735078892

**Comment by victoralmau** [comment]
Changes done.
If everything is correct, do we add the `python-magic` library to https://github.com/Tecnativa/doodba or is it optional?

**Comment by pedrobaeza** [comment]
> If everything is correct, do we add the `python-magic` library to https://github.com/Tecnativa/doodba or is it optional?

IMO we should add it for all (as it affects also to non DMS users that attach a video in chatter), but meanwhile we can add it to current instances through the regular way. cc @Yajo

File: dms/readme/INSTALL.rst [hunk 3]
```diff
@@ -4,3 +4,6 @@ Preview
 `mail_preview_base` is required for DMS but it is recommended to install all
 the other `mail_preview` modules from `social` OCA repository
 in order to improve the preview of files.
+
+`python-magic` library is recommended to install to correctly define file
+types and file preview.
```
Comment (pedrobaeza) [inline dms/readme/INSTALL.rst]:
```suggestion
`python-magic` library is recommended to be installed for having whole support
to get proper file types and file preview.
```

**Review by pedrobaeza — APPROVED** [review]

File: dms/tests/test_file.py [hunk 4]
```diff
@@ -84,3 +90,9 @@ def test_content_file_res_mimetype(self):
         self.assertEqual(file_svg.res_mimetype, "image/svg+xml")
         file_logo = self.env.ref("dms.file_02_demo")
         self.assertEqual(file_logo.res_mimetype, "image/jpeg")
+
+    def test_content_file_res_mimetype_magic_library(self):
+        if not magic:
+            self.skipTest("Without python-magic library installed")
```
Comment (yajo) [inline dms/tests/test_file.py:96]: FTR there's a decorator for this: https://docs.python.org/3/library/unittest.html#unittest.skipIf

However it's OK as it is 😊

**Review by yajo — APPROVED** [review]

**Comment by yajo** [comment]
Failed due to https://github.com/odoo/odoo/pull/75387.

**Comment by rvalyi** [comment]
Hello @Yajo, you could use https://github.com/OCA/l10n-brazil/pull/1587/commits/d43e0eb3329a8116628259a1b6f5ec556e59efb5 
(see https://odoo-community.org/groups/contributors-15/contributors-206982?mode=date&date_begin=2021-08-01&date_end=2021-09-01 )
to workarround the issue meanwhile if you want. That being said it seems Odoo SA is moving forward with this issue so may be waiting they fix it is an option.

**Comment by pedrobaeza** [comment]
/ocabot merge patch

It can be fw-port it to v14 as well

**Pull request merged**